### PR TITLE
mgr/nfs: raise NotADirectoryError instead of its instance

### PR DESCRIPTION
**Pull request by dparmar18**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by dparmar18** [comment]
https://tracker.ceph.com/issues/65265#note-6

technically the code should work, the helper in `utilts.py` raises exception and then the `validate_cephfs_path()` in `export.py` should catch it and raise `NFSException(f"path {path} is not a dir", -errno.ENOTDIR)`. What I think the problem is that the current code raises an instance of `NotADirectoryError` and not the class itself, so for this I'd need to catch it something like `except NotADirectoryError as e:` instead of `except NotADirectoryError:`. I'm switching to raising the class instead of instance, this should make `except NotADirectoryError:` work fine. Will test it out on teuthology(it it is working fine 🥲)

**Comment by dparmar18** [comment]
https://pulpito.ceph.com/dparmar-2024-04-10_06:37:26-fs:nfs-wip-65265-distro-default-smithi/

File: src/pybind/mgr/nfs/utils.py [hunk 1]
```diff
@@ -101,4 +101,4 @@ def _get_cephfs_client() -> CephfsClient:
         stx = fs_handle.statx(path.encode('utf-8'), cephfs.CEPH_STATX_MODE,
                               cephfs.AT_SYMLINK_NOFOLLOW)
         if not stat.S_ISDIR(stx.get('mode')):
-            raise NotADirectoryError()
```
Comment (vshankar) [inline src/pybind/mgr/nfs/utils.py:104]: Are you sure about his? I understand this is still draft, but...

**Comment by dparmar18** [comment]
TL;DR - `no active mgr (MGR_DOWN)` warning is seen before any test case runs + the trace-back mentioned in the tracker are not unhandled failures but just debug logs.

issue doesn't exist in this code, for complete RCA of this please check the comments in https://tracker.ceph.com/issues/65265.

File: src/pybind/mgr/nfs/utils.py (hunk 1 above)
Comment (dparmar18) [reply src/pybind/mgr/nfs/utils.py:104]: https://github.com/ceph/ceph/pull/56759#issuecomment-2058566736
